ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/sftp_outbound_transfer.py

```diff
@@ -302,6 +303,99 @@ def _handle_s3_event(s3_object_key: str):
+                )
+        except botocore.exceptions.ClientError as e:
+            logger.error("Error while polling SQS queue: %s", e)
+        except Exception as e:
+            logger.error("Unexpected error while polling SQS queue: %s", e)
+        finally:
+            # sleep for 5 minutes
+            if not loop_forever:
+                break
+            time.sleep(5 * 60)
+
+
+def handle_sqs_message(s3_message):
+    try:
+        if len(s3_message) == 0:
+            raise ValueError("Invalid inner S3 event with empty records")
+        for s3_record in s3_message["Records"]:
+            s3_event_time = datetime.fromisoformat(
+                s3_record["eventTime"].removesuffix("Z")
+            ).replace(tzinfo=timezone.utc)
+            s3_object_key = unquote(s3_record["s3"]["object"]["key"])
+            s3_event_name = s3_record["eventName"]
+            s3_event_type = S3EventType.from_event_name(s3_event_name)
+            # Append to all future logs information about the S3 Event
+            logger.append_keys(
+                s3_object_key=s3_object_key,
+                s3_event_name=s3_event_name,
+                s3_event_type=s3_event_type.name,
+                s3_event_time=s3_event_time.isoformat(),
+            )
+
+            _handle_s3_event(s3_object_key=s3_object_key)
+
+    except Exception as exc:
+        sns_resource = boto3.resource("sns", config=BOTO_CONFIG)
+
+        notification: StatusNotification
+        if isinstance(exc, BaseTransferError):
+            notification = StatusNotification(
+                details=TransferFailedDetails(
+                    partner=exc.partner or "unknown",
+                    timestamp=datetime.utcnow(),
+                    object_key=exc.s3_object_key,
+                    error_name=exc.__class__.__name__,
+                    reason=exc.message,
+                )
+            )
+
+            topic_arns_by_partner: dict[str, str] = json.loads(SNS_TOPIC_ARNS_BY_PARTNER_JSON)
+            if exc.partner and exc.partner in topic_arns_by_partner:
+                partner_topic = sns_resource.Topic(topic_arns_by_partner[exc.partner])
+                logger.info(
+                    "%s status notification topic configured. Sending error notification",
+                    exc.partner,
+                )
+                send_notification(topic=partner_topic, notification=notification)
+        else:
+            notification = StatusNotification(
+                details=UnknownErrorDetails(
+                    timestamp=datetime.utcnow(), error_name=exc.__class__.__name__, reason=str(exc)
+                )
+            )
+
+        logger.info("Sending error notification to BFD catch-all topic %s", BFD_SNS_TOPIC_ARN)
+        send_notification(topic=sns_resource.Topic(BFD_SNS_TOPIC_ARN), notification=notification)
+
+        raise
+    finally:
+        _, exception, _ = sys.exc_info()
```

Comment:
What's the purpose for checking `exc_info` here? Are we trying to catch a specific subclass of exceptions that are considered unrecoverable?